project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/execution/exception/KotlincExceptionTranslator.kt

```diff
@@ -0,0 +1,128 @@
+package org.plan.research.minimization.plugin.execution.exception
+
+import arrow.core.Either
+import arrow.core.Option
+import arrow.core.getOrElse
+import arrow.core.raise.either
+import arrow.core.raise.ensure
+import arrow.core.raise.ensureNotNull
+import arrow.core.raise.option
+import com.intellij.build.events.BuildEvent
+import com.intellij.build.events.FileMessageEvent
+import com.intellij.build.events.MessageEvent
+import org.jetbrains.kotlin.utils.addToStdlib.indexOfOrNull
+import org.plan.research.minimization.plugin.errors.CompilationPropertyCheckerError
+import org.plan.research.minimization.plugin.model.BuildEventTranslator
+import org.plan.research.minimization.plugin.model.CaretPosition
+
+/**
+ * KotlincExceptionTranslator provides functionality to interpret build events and transform them into either
+ * general compilation errors or internal Kotlin compiler exceptions.
+ */
+class KotlincExceptionTranslator: BuildEventTranslator {
```

Comment:
Great! But we need some tests for this class. It will also show how it actually works